applications/CompressiblePotentialFlowApplication/python_scripts/compute_lift_jump_process_2d.py

```diff
@@ -0,0 +1,55 @@
+import KratosMultiphysics
+import KratosMultiphysics.CompressiblePotentialFlowApplication as CPFApp
+from KratosMultiphysics.CompressiblePotentialFlowApplication.define_wake_process_2d import DefineWakeProcess2D as DefineWakeProcess2D
+
+def Factory(settings, Model):
+    if(not isinstance(settings, KratosMultiphysics.Parameters)):
+        raise Exception(
+            "expected input shall be a Parameters object, encapsulating a json string")
+
+    return ComputeLiftJumpProcess2D(Model, settings["Parameters"])
+
+# TODO Implement this process in C++ and make it open mp parallel to save time selecting the wake elements
+class ComputeLiftJumpProcess2D(DefineWakeProcess2D):
+    def __init__(self, Model, settings):
+        # Call the base Kratos process constructor
+        KratosMultiphysics.Process.__init__(self)
+
+        # Check default settings
+        default_settings = KratosMultiphysics.Parameters(r'''{
+            "model_part_name": "",
+            "velocity_infinity": [1.0,0.0,0],
+            "create_output_file": false
+        }''')
+        settings.ValidateAndAssignDefaults(default_settings)
+
+        body_model_part_name = settings["model_part_name"].GetString()
+        if body_model_part_name == "":
+            err_msg = "Empty model_part_name in DefineWakeProcess2D\n"
+            err_msg += "Please specify the model part that contains the body surface nodes"
+            raise Exception(err_msg)
+        self.body_model_part = Model[body_model_part_name]
+
+        self.velocity_infinity = [0,0,0]
+        self.velocity_infinity[0] = settings["velocity_infinity"][0].GetDouble()
+        self.velocity_infinity[1] = settings["velocity_infinity"][1].GetDouble()
+        self.velocity_infinity[2] = settings["velocity_infinity"][2].GetDouble()
+        self.create_output_file = settings["create_output_file"].GetBool()
+
+    def ExecuteInitialize(self):
+        # Save the trailing edge for further computations
+        self.SaveTrailingEdgeNode()
```

Comment:
Do you need this?

Reply:
@marcnunezc  I just tried your suggestions but I get this error:

Traceback (most recent call last):
  File "MainKratos.py", line 21, in <module>
    simulation.Run()
  File "C:\Kratos\kratos\python_scripts\analysis_stage.py", line 48, in Run
    self.RunSolutionLoop()
  File "C:\Kratos\applications\CompressiblePotentialFlowApplication\python_scripts\potential_flow_analysis.py", line 108, in RunSolutionLoop
    self.FinalizeSolutionStep()
  File "C:\Kratos\kratos\python_scripts\analysis_stage.py", line 141, in FinalizeSolutionStep
    process.ExecuteFinalizeSolutionStep()
  File "C:\Kratos\applications\CompressiblePotentialFlowApplication\python_scripts\compute_lift_jump_process_2d.py", line 43, in ExecuteFinalizeSolutionStep
    node_velocity_potential_te = self.te.GetSolutionStepValue(CPFApp.VELOCITY_POTENTIAL)
**AttributeError: 'ComputeLiftJumpProcess2D' object has no attribute 'te'**
KRATOS TERMINATED WITH ERROR

so the problem I need this "te" attribute to do my calculations.
I both tried with these lines of code:
def ExecuteInitialize(self):
        # Save the trailing edge for further computations
        self.SaveTrailingEdgeNode()

and without,
but still I get errors. Is it maybe another way to do it?

Regarding your suggestions from last week, I implemented all of them except the one regarding these lines of code which we have to discuss with @inigolcanalejo :
`       self.velocity_infinity[0] = settings["velocity_infinity"][0].GetDouble()
        self.velocity_infinity[1] = settings["velocity_infinity"][1].GetDouble()
        self.velocity_infinity[2] = settings["velocity_infinity"][2].GetDouble()`

Reply:
Ok, I see why you needed the DefineWakeProcess. In any case, I wouldn't inherit from it as i think it is not correct conceptually. 

What I would do is to find the trailining edge again in your compute_lift_jump_process. You can loop over the nodes of the "Body" and save the one that is TRAILING_EDGE, which you will use in your ExecuteFinalizeSolutionStep().

Another option would be to modify the DefineWakeProcess so that it creates a new "SubModelPart" that can be directly read from your process. We can do the first option for now to contain everything in our process. 

Something like:
```
def GetTrailingEdgeNode():
     for node in self.model_part.Nodes: 
            if node.GetValue(KratosMultiphysics.TRALIING_EDGE):
                 self.te=node 
                 break




```

Reply:
I implemented and pushed the first option to avoid inheriting from DefineWakeProcess and it works!
I hope the way I did it is correct, I just added the loop into my ExecuteFinalizeSolutionStep() so I didn't create a new function for that. You can have a look if you want @marcnunezc  and give me feedback.